### PR TITLE
feat: allow toggle slack app to use socket mode

### PR DESCRIPTION
**Pull request by Shifan-Gu**
Closes: 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/clients/Slack/Slackbot.ts [hunk 1]
```diff
@@ -74,27 +74,42 @@ export class SlackBot {
...
-                logLevel: LogLevel.INFO,
-                app: expressApp,
-            });
-
-            const app = new App({
-                ...slackOptions,
-
-                receiver: slackReceiver,
-            });
+            if (this.lightdashConfig.slack?.appToken) {
```
Comment (ZeRego) [inline packages/backend/src/clients/Slack/Slackbot.ts:77]: @Shifan-Gu Let's introduce a `SLACK_SOCKET_MODE` env var and use it for this condition. Two reasons for this: explicit configuration, prevent backwards compatibility issues as old instances might still have `SLACK_APP_TOKEN` env var.

File: packages/backend/src/clients/Slack/Slackbot.ts [hunk 2]
```diff
@@ -74,27 +74,42 @@ export class SlackBot {
...
+                const app = new App({
+                    ...slackOptions,
+                    installationStore: {
+                        storeInstallation: (i) =>
+                            this.slackAuthenticationModel.createInstallation(i),
+                        fetchInstallation: (i) =>
+                            this.slackAuthenticationModel.getInstallation(i),
```
Comment (ZeRego) [inline packages/backend/src/clients/Slack/Slackbot.ts:84]: Can we handle the delete here?
```
                        deleteInstallation: (i) =>
                            this.slackAuthenticationModel.deleteInstallation(i),
```

File: packages/backend/src/clients/Slack/Slackbot.ts (hunk 1 above)
Comment (Shifan-Gu) [reply packages/backend/src/clients/Slack/Slackbot.ts:77]: sure, will do 👍

File: packages/backend/src/clients/Slack/Slackbot.ts (hunk 2 above)
Comment (Shifan-Gu) [reply packages/backend/src/clients/Slack/Slackbot.ts:84]: yes, its missing and should be in place, good catch 👍

**Review by ZeRego — APPROVED** [review]

**Pull request merged**